src/hooks/use-broadcast-channel.ts

```diff
@@ -17,19 +17,16 @@ export default function useBroadcastChannel(
   onMessage: MessageCallback = noop,
   deps = []
 ) {
-  const [broadcasterId] = useState(() => ulid());
-  const [channel, setChannel] = useState(
-    () => new BroadcastChannel(channelName)
-  );
+  const broadcasterId = useRef(() => ulid());
+  const channel = useRef<BroadcastChannel>();
 
   useEffect(() => {
-    if (channel.name !== channelName) {
-      setChannel(() => new BroadcastChannel(channelName));
-    }
+    channel.current = new BroadcastChannel(channelName);
```

Comment:
I don't think we really care about channel name check. Any time `channelName` changes, let's tear down the old channel and create a new one.